content/blog/nanoparquet-0-4-0/index.Rmd

```diff
@@ -0,0 +1,230 @@
+facilitates great features, like `append_parquet()` and the new
+`col_select` argument. (More to come!) The new reader is also much faster,
+see the "Benchmarks" chapter.
+
+## Read a subset of columns
+
+`read_parquet()` now has a new argument called `col_select`, that lets you
+read a subset of the columns from the Parquet file. Unlike for row oriented
+file formats like CSV, this means that the reader never needs to touch the
+columns that are not needed for analysis. Reading a handful of columns only
+always takes the same amount of time, no matter how many more columns the
+Parquet file might have!
+
+You can either use column indices or column names to specify the columns
+to read. Here is an example.
+
+```{r setup}
+library(nanoparquet)
+library(pillar)
+```
+
+```{r include = FALSE}
+if (!file.exists("flights.parquet")) {
+  write_parquet(nycflights13::flights, "flights.parquet")
+}
+```
+
+This is the `nycflights13::flights` data set:
+
+```{r col_select}
+read_parquet(
+  "flights.parquet",
+  col_select = c("dep_time", "arr_time", "carrier")
+)
+```
+
+Use `read_parquet_schema()` if you want to see the structure of the Parquet
+file first:
+
+```{r read_parquet_schema}
+read_parquet_schema("flights.parquet")
+```
+
+The output of `read_parquet_schema()` also shows you the R type that
+nanoparquet will use for each column.
+
+## Appending to Parquet files
+
+The new `append_parquet()` function makes it easy to append new data to
+a Parquet file, without first reading the whole file into memory.
+The schema of the file and the schema new data must match of course. Lets
+merge `nycflights13::flights` and `nycflights23::flights`:
+
+```{r append_parquet}
+file.copy("flights.parquet", "allflights.parquet", overwrite = TRUE)
+append_parquet(nycflights23::flights, "allflights.parquet")
+```
+
+`read_parquet_info()` returns the most basic information about a Parquet
+file:
+
+```{r read_parquet_info}
+read_parquet_info("flights.parquet")
+read_parquet_info("allflights.parquet")
+```
+
+Note that you should probably still create a backup copy of the original
+file when using `append_parquet()`. If the appending process is interrupted
+by a power failure, then you might end up with an incomplete and invalid
+Parquet file.
+
+## Schemas and type conversions
+
+In nanoparquet 0.4.0 `write_parquet()` takes a `schema` argument that
+can customize the R to Parquet type mappings. E.g. by default
+`write_parquet()` writes an R character vector as a `STRING` Parquet type.
+If you'd like to write a certain character column as an `ENUM` type
+instead, you'll need to specify that in `schema`:
+
+```{r schema}
+write_parquet(
+  nycflights13::flights,
+  "newflights.parquet",
+  schema = parquet_schema(carrier = "ENUM")
+)
+read_parquet_schema("newflights.parquet")
+```
+
+Here we wrote the `carrier` column as `ENUM`, and left the other other
+columns to use the default type mappings.
+
+See the [`nanoparquet-types`](
```

Comment:
BTW you should be able to write `?nanoparquet-types` and downlit will autolink.

Reply:
That does not work for me, probably because of the dash.